src/main/java/eu/europeana/metis/sandbox/service/workflow/TransformationServiceImpl.java

```diff
@@ -1,119 +1,131 @@
-    } catch (TransformationException e) {
-      throw new RecordProcessingException(identifier, e);
-    } finally {
-      closeStream(xsltContentInputStream);
+    @Override
+    public RecordInfo transformToEdmInternal(Record recordToTransform) {
+        requireNonNull(recordToTransform, "Record must not be null");
+
+        final byte[] recordTransformed;
+        try {
+            final EuropeanaGeneratedIdsMap europeanaGeneratedIdsMap = new EuropeanaIdCreator()
+                    .constructEuropeanaId(recordToTransform.getContentInputStream(), recordToTransform.getDatasetId());
+            XsltTransformer transformer = getTransformer(getJoinDatasetIdDatasetName(recordToTransform),
+                    recordToTransform.getCountry().xmlValue(), recordToTransform.getLanguage().name().toLowerCase());
+            recordTransformed = transformer
+                    .transformToBytes(recordToTransform.getContent(), europeanaGeneratedIdsMap);
+        } catch (TransformationException | EuropeanaIdException e) {
+            throw new RecordProcessingException(recordToTransform.getProviderId(), e);
+        }
+
+        return new RecordInfo(Record.from(recordToTransform, recordTransformed));
     }
 
-    return resultRecord;
-  }
+    @Override
+    public RecordInfo transform(Record recordToTransform) {
+        InputStream xsltContent = new ByteArrayInputStream(
+                datasetRepository.getXsltContentFromDatasetId(Integer.parseInt(recordToTransform.getDatasetId()))
+                        .getBytes(StandardCharsets.UTF_8));
+        return new RecordInfo(Record.from(recordToTransform, transform(String.valueOf(recordToTransform.getRecordId()), xsltContent,
+                recordToTransform.getContent())));
+    }
+
+    @Override
+    public byte[] transform(String identifier, InputStream xsltContentInputStream,
+                            byte[] recordContent) {
+
+        final byte[] resultRecord;
+        try {
+            XsltTransformer transformer = getNewTransformerObject(identifier, xsltContentInputStream);
+            resultRecord = transformer.transformToBytes(recordContent, null);
+        } catch (TransformationException e) {
+            throw new RecordProcessingException(identifier, e);
+        } finally {
+            closeStream(xsltContentInputStream);
+        }
+
+        return resultRecord;
+    }
+
+    private void closeStream(Closeable closeable) {
+        if (closeable != null) {
+            try {
+                closeable.close();
+            } catch (IOException e) {
+                LOGGER.error("Unable to close transform stream", e);
+            }
+        }
+    }
+
+    private XsltTransformer getTransformer(String datasetName, String edmCountry,
+                                           String edmLanguage) throws TransformationException {
+
+        if (lockRepository.isAcquired(TransformXsltRepository.LOCK_NAME_KEY)) {
```

Comment:
This is not really necessary here.
The xslt will be added during the first ever deployment of the application.
From then onwards there would be updates of it, so the transformation would get the one existent or the one that just replaced the older one.
So such a check here is redundant and would make records throw exceptions(rarely though), for no real reason.